scripts/vector_genotype_concordance/scripts/vector_genotype_concordance.py

```diff
@@ -0,0 +1,363 @@
+                           default=['2L', '2R', '3L', '3R', 'X'], help='The list of chromosomes')
+        count.add_argument('--filter-catalog-url', dest='url', required=False,
+                           default='https://malariagen.github.io/intake/gcs.yml',
+                           help='The url of the catalog containing the filters')
+        count.add_argument('--filter-path', '-f', dest='path', required=False,
+                           default='ag3.site_filters_dt_20200416_gamb_colu_arab',
+                           help='The path in the catalog to the filter required, '
+                                'ie ag3.site_filters_dt_20200416_gamb_colu_arab')
+        count.set_defaults(command=Commands.COUNT)
+        return self
+
+    def with_summarize(self):
+        count = self.subparsers.add_parser("summarize", help='Summarize results of several count output')
+        count.add_argument('--inputs', '-i', nargs='+', dest='inputs', required=True,
+                           help='The list of files to summarize')
+        count.add_argument('--output', '-o', dest='output', required=False, default=None,
+                           help='The output file, by default prints to stdout')
+        count.set_defaults(command=Commands.SUMMARIZE)
+        return self
+
+    def build(self):
+        return self.parser
+
+    def get_arguments(self):
+        return dict(vars(self.parser.parse_args()))
+
+
+class Summarizer:
+    """Summarizer manipulates the statistics collected and prints concordance based on various groupings"""
+
+    @staticmethod
+    def to_dataframe(stream):
+        return pandas.read_csv(stream, dtype={SAMPLE_HEADER: 'str', CHROMOSOME_HEADER: 'str', CONTROL_HEADER: 'str',
+                                              TEST_HEADER: 'str', COUNT: 'int64'})
+
+    def __init__(self, dataframes):
+        self.dataframe = pandas.concat(dataframes, ignore_index=True, sort=False)
+        self.dataframe['mismatch'] = self.dataframe[CONTROL_HEADER] != self.dataframe[TEST_HEADER]
+        self.dataframe['category'] = self.dataframe[[CONTROL_HEADER, TEST_HEADER]].agg('_'.join, axis=1)
+
+    def compute_total_concordance(self):
+        total = self._compute_total_concordance()
+        no_missing_homref_homref = self._compute_concordance_no_missing_no_homref_homref()
+        result = pandas.concat([total, no_missing_homref_homref], ignore_index=True, sort=False)
+        result['description'] = ['Total', 'Total no missing no homref_homref']
+        return result.reindex(columns=['description', 'total', 'match', 'concordance'])
+
+    def compute_concordance_per_category(self):
+        return self._concordance_per_category(self.dataframe)
+
+    def compute_concordance_per_sample_category(self):
+        return self.dataframe.groupby([TEST_HEADER, CONTROL_HEADER, SAMPLE_HEADER]) \
+            .agg(count=('count', sum)) \
+            .groupby(level=2) \
```

Comment:
Haven't tried, but can you use the column name instead of the numerical index for the groupby()?  Would make it easier to know that you are grouping again by sample.

Reply:
I agree it's not super readable.  I tried giving the name, but ended up with a key error (key already exist).